src/test/scala/afenton/bazel/bsp/LspTestProcess.scala

```diff
@@ -204,13 +204,17 @@ case class LspTestProcess(workspaceRoot: Path):
       openRequests <- Ref.of[IO, OpenRequests](Map.empty)
       counter <- Ref.of[IO, Int](1)
       client = BspClient(openRequests, bspInQ, counter)
+      exitSwitch <- Deferred[IO, Either[Throwable, Unit]]
       result = IO.both(
-        processActions(client, actions),
-        (processBspOut(
-          Stream.fromQueueUnterminated(bspOutQ),
-          openRequests,
-          duration
-        ) <& processBspErr(Stream.fromQueueUnterminated(bspErrQ), duration))
+        processActions(client, actions, exitSwitch),
+        IO.race(
+          processBspOut(
+            Stream.fromQueueUnterminated(bspOutQ),
+            openRequests,
+            exitSwitch
+          ),
+          processBspErr(Stream.fromQueueUnterminated(bspErrQ))
+        ).map(_.left.get)
```

Comment:
I don't love the `left.get` but honestly at first glance I can't see all what is happening here.

Reply:
Yeah, need some help simplifying all this. Check your slack :)